src/ai-services/ant-eater.service.ts

```diff
@@ -0,0 +1,127 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { BaseAssistantService } from './base-assistant.service';
+import { MessageService } from 'src/message/message.service';
+import { dynamicImport } from 'src/utils/dynamic-import.utils';
+import * as Console from "console";
+import {Message} from "../message/message.entity";
+import {CreateMessageDTO} from "../message/dto/create-message.dto";
+import {AIMessage} from "langchain/schema";
+@Injectable()
+export class AntEaterService extends BaseAssistantService {
+    modelName = 'antEater';
+    description = `AntEater is an AI assistant that has the ability to summarize conversation history, while also having a buffer of recent conversations 
+    and holding a cache of messages so previous messages dont have to be grabbed on every instance.`;
+
+    private chatHistoryCache = new Map<number, Array<any>>();
```

Comment:
This is not the right way to cache messages! This is just a map, and it does not have any eviction policy. You keep adding all messages to this map and the memory will run out. Not only will this not speed up the process, but it will also cause the server to crash. Moreover, there is already an example of using cache: see `continueConversation` operation in `src/conversation/conversation.service.ts`